src/columns.jl

```diff
@@ -181,21 +188,23 @@ julia> ncols(ndsparse(d, [7,8,9]))
 """
 function ncols end
 ncols(c::Columns) = nfields(typeof(c.columns))
+ncols(c::Columns{<:Pair, <:Pair}) = ncols(c.columns.first) => ncols(c.columns.second)
 ncols(c::AbstractArray) = 1
 
 size(c::Columns) = (length(c),)
 Base.IndexStyle(::Type{<:Columns}) = IndexLinear()
 summary(c::Columns{D}) where {D<:Tuple} = "$(length(c))-element Columns{$D}"
 
 empty!(c::Columns) = (foreach(empty!, c.columns); c)
+empty!(c::Columns{<:Pair, <:Pair}) = (foreach(empty!, c.columns.first.columns); foreach(empty!, c.columns.second.columns); c)
 
 function similar(c::Columns{D,C}) where {D,C}
-    cols = map(similar, c.columns)
+    cols = map_pair(similar, c.columns)
     Columns{D,typeof(cols)}(cols)
 end
 
 function similar(c::Columns{D,C}, n::Integer) where {D,C}
-    cols = map(a->similar(a,n), c.columns)
+    cols = map_pair(a->similar(a,n), c.columns)
```

Comment:
maybe this should be called `_map` (this is the convention we use when we are avoiding piracy (e.g. _promote_op)). `map_pair` seems like it only works on pairs or something.

Reply:
Good point, I'll change that.